modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/matrix/stats/RunningStats.java

```diff
@@ -0,0 +1,358 @@
+            for (String cFieldName : cFieldNames) {
+                if (cFieldVals.containsKey(cFieldName) == true) {
+                    newVal = cFieldVals.get(cFieldName) + 1.0 / (docCount * (docCount - 1.0)) * dR * deltas.get(cFieldName);
+                    cFieldVals.put(cFieldName, newVal);
+                } else {
+                    cFieldVals.put(cFieldName, 0.0);
+                }
+            }
+            if (cFieldVals.size() > 0) {
+                covariances.put(fieldName, cFieldVals);
+            }
+        }
+    }
+
+    /**
+     * Merges the descriptive statistics of a second data set (e.g., per shard)
+     *
+     * running computations taken from: http://prod.sandia.gov/techlib/access-control.cgi/2008/086212.pdf
+     **/
+    public void merge(final RunningStats other) {
+        if (other == null) {
+            return;
+        }
+        final double nA = docCount;
+        final double nB = other.docCount;
+        // merge count
+        docCount += other.docCount;
+
+        final HashMap<String, Double> deltas = new HashMap<>();
+        double meanA, varA, skewA, kurtA, meanB, varB, skewB, kurtB;
+        double d, d2, d3, d4, n2, nA2, nB2;
+        double newSkew, nk;
+        // across fields
+        for (Map.Entry<String, Double> fs : other.means.entrySet()) {
+            final String fieldName = fs.getKey();
+            meanA = means.get(fieldName);
+            varA = variances.get(fieldName);
+            skewA = skewness.get(fieldName);
+            kurtA = kurtosis.get(fieldName);
+            meanB = other.means.get(fieldName);
+            varB = other.variances.get(fieldName);
+            skewB = other.skewness.get(fieldName);
+            kurtB = other.kurtosis.get(fieldName);
+
+            // merge counts of two sets
+            counts.put(fieldName, counts.get(fieldName) + other.counts.get(fieldName));
+
+            // merge means of two sets
+            means.put(fieldName, (nA * means.get(fieldName) + nB * other.means.get(fieldName)) / (nA + nB));
+
+            // merge deltas
+            deltas.put(fieldName, other.fieldSum.get(fieldName) / nB - fieldSum.get(fieldName) / nA);
+
+            // merge totals
+            fieldSum.put(fieldName, fieldSum.get(fieldName) + other.fieldSum.get(fieldName));
+
+            // merge variances, skewness, and kurtosis of two sets
+            d = meanB - meanA;          // delta mean
+            d2 = d * d;                 // delta mean squared
+            d3 = d * d2;                // delta mean cubed
+            d4 = d2 * d2;               // delta mean 4th power
+            n2 = docCount * docCount;   // num samples squared
+            nA2 = nA * nA;              // doc A num samples squared
+            nB2 = nB * nB;              // doc B num samples squared
+            // variance
+            variances.put(fieldName, varA + varB + d2 * nA * other.docCount / docCount);
```

Comment:
This caught my eye at first because I wasn't sure how the order of operations was falling out, then I was curious.  Is the usage of delta squared mean here equivalent to a Pooled Variance like http://www.emathzone.com/tutorials/basic-statistics/combined-variance.html ?

I was expecting something like:

```
(
  nA * (Math.pow(varA,2) + Math.pow(meanA - combinedMean, 2))
  +
  nB * (Math.pow(varB,2) + Math.pow(meanB - combinedMean, 2))
) / docCount
```

But my math skills are *super rusty*, so heaps of salt etc etc :)